### ISSUE TITLE
Issue with sphinx 4.3.x

### ISSUE THREAD

**Issue by garf365** [domdfcoding/enum_tools#45]
Hello everyone,

enum_tools version: 0.6.5

After updating Sphinx version from 4.2.0 to 4.3.1, I encounter this issue:

```
# Sphinx version: 4.3.1
# Python version: 3.9.9 (CPython)
# Docutils version: 0.16 release
# Jinja2 version: 3.0.3
# Last messages:
#   building [mo]: targets for 0 po files that are out of date
#   building [html]: targets for 5 source files that are out of date
#   updating environment:
#   [new config]
#   5 added, 0 changed, 0 removed
#   reading sources... [ 20%] configuration
#   reading sources... [ 40%] examples
#   reading sources... [ 60%] index
#   reading sources... [ 80%] install
#   reading sources... [100%] modules
# Loaded extensions:
#   sphinx.ext.mathjax (4.3.1) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/ext/mathjax.py
#   sphinxcontrib.applehelp (1.0.2) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinxcontrib/applehelp/__init__.py
#   sphinxcontrib.devhelp (1.0.2) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinxcontrib/devhelp/__init__.py
#   sphinxcontrib.htmlhelp (2.0.0) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinxcontrib/htmlhelp/__init__.py
#   sphinxcontrib.serializinghtml (1.1.5) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinxcontrib/serializinghtml/__init__.py
#   sphinxcontrib.qthelp (1.0.3) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinxcontrib/qthelp/__init__.py
#   alabaster (0.7.12) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/alabaster/__init__.py
#   sphinx.ext.autodoc.preserve_defaults (1.0) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/ext/autodoc/preserve_defaults.py
#   sphinx.ext.autodoc.type_comment (4.3.1) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/ext/autodoc/type_comment.py
#   sphinx.ext.autodoc (4.3.1) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py
#   sphinx.ext.doctest (4.3.1) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/ext/doctest.py
#   sphinx.ext.graphviz (4.3.1) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/ext/graphviz.py
#   enum_tools.autoenum (0.6.5) from /home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/enum_tools/autoenum.py
Traceback (most recent call last):
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/application.py", line 344, in build
    self.builder.build_update()
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 294, in build_update
    self.build(to_build,
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 308, in build
    updated_docnames = set(self.read())
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 415, in read
    self._read_serial(docnames)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 436, in _read_serial
    self.read_doc(docname)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 476, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/io.py", line 189, in read_doc
    pub.publish()
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/core.py", line 217, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/io.py", line 109, in read
    self.parse()
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/readers/__init__.py", line 77, in parse
    self.parser.parse(self.input, document)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/parsers.py", line 101, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 170, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2342, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/ext/autodoc/directive.py", line 162, in run
    documenter.generate(more_content=self.content)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 984, in generate
    self.document_members(all_members)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 860, in document_members
    documenter.generate(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/enum_tools/autoenum.py", line 278, in generate
    self.document_members(all_members)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/enum_tools/autoenum.py", line 139, in document_members
    self._do_document_members(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/enum_tools/autoenum.py", line 224, in _do_document_members
    documenter.generate(
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/enum_tools/autoenum.py", line 344, in generate
    ret = begin_generate(self, real_modname, check_module)
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/sphinx_toolbox/utils.py", line 394, in begin_generate
    if not documenter.import_object():
  File "/home/gilles/innova/dev/python-client/.tox/docs/lib/python3.9/site-packages/enum_tools/autoenum.py", line 324, in import_object
    self._datadescriptor = False
AttributeError: can't set attribute
```

With same version of enum_tools (0.6.5), I got this error with sphinx 4.3.0 and 4.3.1 but not with 4.2.0

**Comment by domdfcoding** [comment]
Sorry about this. `enum_tools` hadn't been updated to account for changes made in 3.5 and later. #48 should resolve this

@garf365 @1kastner can you try that branch with Sphinx 4.3 and let me know if it works for you?

You can install the branch with ``pip install git+https://github.com/domdfcoding/enum_tools@sphinx4``

**Comment by calebstewart** [comment]
I'm not the original author of this issue, but I had the same issue, and the branch does fix the problem for me :+1:

**Comment by 1kastner** [comment]
Everything looks good to me!

**Comment by domdfcoding** [comment]
Thanks both for your feedback. I will merge the PR and make a release.

**Comment by domdfcoding** [comment]
I've released version [0.7.0](https://pypi.org/project/enum-tools/0.7.0/) with support for Sphinx 4.x

**Comment by garf365** [comment]
Thank you !
